cli/internal/env/env.go

```diff
@@ -91,44 +92,55 @@ func fromKeys(all EnvironmentVariableMap, keys []string) EnvironmentVariableMap
 	return output
 }
 
-// fromPrefixes returns a map of env vars and their values based on include/exclude prefixes
-func fromPrefixes(all EnvironmentVariableMap, includes []string, exclude string) EnvironmentVariableMap {
+func fromMatching(all EnvironmentVariableMap, keyMatchers []string, excludePrefix string) (EnvironmentVariableMap, error) {
 	output := EnvironmentVariableMap{}
-	for _, prefix := range includes {
+	compileFailures := []string{}
+
+	for _, keyMatcher := range keyMatchers {
+		rex, err := regexp.Compile(keyMatcher)
+		if err != nil {
+			compileFailures = append(compileFailures, keyMatcher)
+			continue
+		}
+
 		for k, v := range all {
-			// Skip vars that have the exclude prefix
-			if exclude != "" && strings.HasPrefix(k, exclude) {
+			// we can skip the keys that match the excludedPrefix
+			if excludePrefix != "" && strings.HasPrefix(k, excludePrefix) {
 				continue
 			}
 
-			// if it has the prefix, include it
-			if strings.HasPrefix(k, prefix) {
+			if rex.Match([]byte(k)) {
 				output[k] = v
 			}
 		}
 	}
-	return output
+
+	if len(compileFailures) > 0 {
+		return nil, fmt.Errorf("The following env prefixes failed to compile to regex: %s", strings.Join(compileFailures, ", "))
+	}
+
+	return output, nil
 }
 
 // GetHashableEnvVars returns all sorted key=value env var pairs for both frameworks and from envKeys
-func GetHashableEnvVars(keys []string, prefixes []string) DetailedMap {
+func GetHashableEnvVars(keys []string, matchers []string, excludePrefix string) (DetailedMap, error) {
```

Comment:
I'm a little confused as to the difference between the `excludePrefix` passed here versus the `excludePrefix` passed to `fromMatching`. It appears that the former is the key while the latter is the value in the environment variable map. Could we make that clear in the naming?

Reply:
yeah, I didn't want to spend _too_ much time getting this designed right. I've pushed up a commit that hopefully makes it a little more clear and makes `fromMatching` not as coupled to how it's used